src/Data/Types/Interval.php

```diff
@@ -0,0 +1,37 @@
+<?php
+namespace LORIS\Data\Types;
+
+/**
+ * An Interval data type represents a time interval or duration.
+ *
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ */
+class Interval implements \LORIS\Data\Type
```

Comment:
Shouldn't this be an abstract class with a constructor that takes 2 `\LORIS\Data\Type` of the same class?
We would have a DateInterval, TimeInterval? 
I am not sure how or where this is meant to be used.

Reply:
What's the difference between a date interval and a time interval?.. I think I originally called it "Duration" but after using it for candidate age an age being a duration didn't quite sit right with me.

Reply:
I would expect the unit to be different. 
The constructor would take 2 dates, 2 time or 2 whatever.

Reply:
Why would it take 2 anything? This is the data type, not the instance of the data. The instance of the data can be whatever magnitude (day, time, whatever) that it wants.

Reply:
Maybe I misunderstand the meaning of "Interval". In my mind, it is the distance measured in "unit" between two "things".

Reply:
@driusan i guess i'm confused too, if that's the case why does enumeration take a list of options if its the datatype not the instance ?

it seems like the min/max for interval is equivalent to options for enumeration

(notice here im not comparing to the maxsize in `integer` because that seems to be a meta-value required to define the size of the int rather than the int itself)

Reply:
An Interval is the time between 2 points in time. For instance: the data type for candidate age or an instrument question asking "how long do you sleep each night?" There's no start/end point, there's just a duration of time. It can be days, minutes, seconds, or years, that's just a question of magnitude, not type.

I don't know what min/max you're talking about for intervals?

The constructors take the amount of parameters they need to describe the type. An enumeration data type needs to know what the options are because it's a part of the type. Two enumerations with different options aren't the same. A string needs to know the max size to know the SQL data type that it corresponds to. The integer should have the `maxsize` removed (at least for now), it looks like it's not used and I think it's just leftover from basing the file on the string type initially (but later it might be a good idea to add the ability to define constraints such as max or min size..)

Reply:
To make things concrete, looking at #6936 Interval is the data type currently used for Candidate Age and Window Difference in instruments. How would you describe the data type of those fields?

Reply:
Looking at https://en.wikipedia.org/wiki/ISO_8601#Durations and comparing the ISO8601 definition of "Time Interval" and "Duration" it might be better to go back to calling them "Durations", but it doesn't feel right calling a window difference or an candidate age a duration to me.. I'm open to any other suggestions.

Reply:
I guess the confusion for me stemmed from the fact that I thought the "interval" was computed, in this function, from 2 days, 2 times, 2 whatever (thus the min/max values i mentioned previously). I also did not necessarily associate an interval with only points in time... and interval can be [a space between objects, units, points, or states](https://www.merriam-webster.com/dictionary/interval).

Thinking of it now the interval/duration you are talking about is basically an int/decimal with a unit. does it need it's own data type? isnt the candidate age and int of years, months, days... ? 

When i hear "interval" i automatically think of 2 values... not 1 value with a unit so it might just be a terminoology issue

Reply:
Okay, I've switched it from Interval to Duration. I don't know if it makes sense to extend from a programmatic perspective (they don't really have much behaviour to inherit and if they do I'm not sure if they'll be more or less likely to be the same..) but I made it extend Decimal.

Reply:
I think it makes more sens like that. and yeah you'r right there isn't much to inherit.

Reply:
I took away the extends because the more I think about it the more I think it might be a limiting assumption. For instance, the example in the above ISO8601 duration:

```
For example, "P3Y6M4DT12H30M5S" represents a duration of "three years, six months, four days, twelve hours, thirty minutes, and five seconds". 
```

That *could* be represented as a whole number of seconds, but it's not a very natural way of doing it.